### PR TITLE
VIM-4087: Load Trailers for VOD items

### PR DESCRIPTION
**Pull request by nicolelehrer**
#### Ticket
**Required for Vimeans only**
[VIM-4087](https://vimean.atlassian.net/browse/VIM-4087)

#### Ticket Summary
We need to access the VOD trailer URI from a VIMVideo.

#### Implementation Summary
Function prototype was added to `VIMVideo+VOD` header for method `vodTrailerURI` so it would be publicly accessible.

#### How to Test
NA

### CODE REVIEW

**Comment by huebnerob** [comment]
Looks good 👍

**Comment by nicolelehrer** [comment]
Thanks Rob, sorry for the additional changes post review - I needed to add 2 more changes
1 - Augment our response serialization list to accept video from onDemand pages, and also 
2 - When we need to get videos back from onDemand pages, we need to use a video override flag in the request to get valid video objects back. So now we have another request video method that includes this override in the request parameters. 

Ready for another look, thanks @huebnerob

File: VIMNetworking/Networking/VimeoClient/VIMClient.m [hunk 1]
```diff
@@ -264,6 +264,17 @@ - (instancetype)initWithBaseURL:(NSURL *)url
     return [self requestDescriptor:descriptor completionBlock:completionBlock];
 }
 
+- (id<VIMRequestToken>)vodVideoWithURI:(NSString *)URI completionBlock:(VIMRequestCompletionBlock)completionBlock
+{
+    VIMRequestDescriptor *descriptor = [[VIMRequestDescriptor alloc] init];
+    descriptor.urlPath = URI;
+    descriptor.modelClass = [VIMVideo class];
+    descriptor.modelKeyPath = @"";
```
Comment (huebnerob) [inline VIMNetworking/Networking/VimeoClient/VIMClient.m]: nit: we can just leave this off

**Comment by huebnerob** [comment]
Looks great @nicolelehrer 👍

**Pull request merged**